### PR TITLE
fix: react hooks can not in condition segment

### PR DESCRIPTION
**Pull request by Innei**
### Description

Fix React Hooks usage.

### CODE REVIEW

File: src/hooks.ts [hunk 1]
```diff
@@ -14,24 +14,36 @@ export function useTargetLanguages() {
...
+  React.useEffect(() => {
+    if (!preferences.getSystemSelection) {
+      return
+    }
+    getSelectedText()
+      .then((cbText) => {
+        if (isMountedRef.current) setText((cbText ?? '').trim())
+      })
+      .catch(() => {})
+  }, [preferences.getSystemSelection])
```
Comment (SukkaW) [inline src/hooks.ts:46]: IMHO this is also wrong. Here is a way to implement this without `useRef`:

```suggestion
  React.useEffect(() => {
    if (!preferences.getSystemSelection) {
      return
    }
    let isCancelled = false;
    getSelectedText()
      .then((cbText) => {
        if (isCancelled) setText((cbText ?? '').trim())
      })
      .catch(() => {})
      
    return () => { isCancelled = true };
  }, [preferences.getSystemSelection])
```

Comment (Innei) [reply src/hooks.ts]: Learned it!

Comment (SukkaW) [reply src/hooks.ts]: @Innei 

This known pattern can also help avoiding the following race condition:

```tsx
interface ExampleComponentProps {
  dataKey: 'data1' | 'data2'
}

const ExampleComponent = ({ dataKey }: ExampleComponentProps) => {
  const [data, setData] = useState(null);
  useEffect(() => {
    let isCancelled = false;

    someAsyncStuff().then(data => {
      if (!isCancelled) {
        setData(data);
      }
    });

    return () => {
      isCancelled = true;
    };
  }, [dataKey]);
}
```

```
│ Request data 1 ────────────────────────────────────────► data1 response (setData(data1)) │
           │ Request data 2 ────► data2 response (setData(data2)) │
```

`useIsMountedRef` can not help with that.

Comment (Innei) [reply src/hooks.ts]: Learned a lot!

**Pull request merged**